daemon/inertiad/webhook/webhook.go

```diff
@@ -0,0 +1,49 @@
+package webhook
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// Payload represents a generic webhook payload
+type Payload interface {
+	GetEventType() string
+	GetRepoName() string
+	GetRef() string
+	GetGitURL() string
+	GetSSHURL() string
+}
+
+// Parse takes in a webhook request and parses it into one of the supported types
+func Parse(r *http.Request) (Payload, error) {
```

Comment:
Generally in the daemon, all code that writes take an `io.Writer` so that we can configure where stuff goes, then:

```go
fmt.Printf(out, "Parsing webhook...")
```

Reply:
Example: https://github.com/ubclaunchpad/inertia/blob/master/daemon/inertiad/project/deployment.go#L143